tests/test_manager/test_manager.py

```diff
@@ -585,6 +587,75 @@ class TestMultiAgentManagerThreadedModeWithPassword(BaseTestMultiAgentManager):
+    - 'fetchai/p2p_libp2p' of type connection: the new version '0.20.0' conflicts with existing version '0.21.0' of the same package required by agents: [<fetchai/weather_station:0.27.0>]
+    - 'fetchai/soef' of type connection: the new version '0.21.0' conflicts with existing version '0.22.0' of the same package required by agents: [<fetchai/weather_station:0.27.0>]
+    - 'fetchai/contract_api' of type protocol: the new version '0.14.0' conflicts with existing version '1.0.0' of the same package required by agents: [<fetchai/weather_station:0.27.0>]
+    - 'fetchai/default' of type protocol: the new version '0.15.0' conflicts with existing version '1.0.0' of the same package required by agents: [<fetchai/weather_station:0.27.0>]
+    - 'fetchai/fipa' of type protocol: the new version '0.16.0' conflicts with existing version '1.0.0' of the same package required by agents: [<fetchai/weather_station:0.27.0>]
+    - 'fetchai/ledger_api' of type protocol: the new version '0.13.0' conflicts with existing version '1.0.0' of the same package required by agents: [<fetchai/weather_station:0.27.0>]
+    - 'fetchai/oef_search' of type protocol: the new version '0.16.0' conflicts with existing version '1.0.0' of the same package required by agents: [<fetchai/weather_station:0.27.0>]
+    - 'fetchai/signing' of type protocol: the new version '0.13.0' conflicts with existing version '1.0.0' of the same package required by agents: [<fetchai/weather_station:0.27.0>]
+    - 'fetchai/state_update' of type protocol: the new version '0.13.0' conflicts with existing version '1.0.0' of the same package required by agents: [<fetchai/weather_station:0.27.0>]
+    """
+    )
+
+    def setup(self):
+        """Set up the test case."""
+        self.project_public_id = MY_FIRST_AEA_PUBLIC_ID
+        self.tmp_dir = TemporaryDirectory()
+        self.working_dir = os.path.join(self.tmp_dir.name, "MultiAgentManager_dir")
+        self.project_path = os.path.join(
+            self.working_dir, self.project_public_id.author, self.project_public_id.name
+        )
+        assert not os.path.exists(self.working_dir)
+        self.manager = MultiAgentManager(self.working_dir)
+
+    def test_run(self):
+        """
+        Run the test.
+
+        First add agent project "fetchai/weather_station:0.27.0",
+        and then add "fetchai/weather_client:0.27.0".
+        The second addition will fail because the projects
+        contain conflicting AEA package versions.
+        """
+        self.manager.start_manager()
+        weather_station_id = PublicId.from_str("fetchai/weather_station:0.27.0")
+        self.manager.add_project(weather_station_id)
+        weather_client_id = PublicId.from_str("fetchai/weather_client:0.27.0")
+        with pytest.raises(
+            ProjectPackageConsistencyCheckError,
+            match=re.escape(self.EXPECTED_ERROR_MESSAGE),
+        ):
+            self.manager.add_project(weather_client_id)
```

Comment:
this is the code to reproduce issue https://github.com/fetchai/agents-aea/issues/2459